examples/llm_compression/torch/qat_with_nls_downstream/main.py

```diff
@@ -0,0 +1,773 @@
+                selected_rank = int(np.random.choice(lora_rank_space))
+            else:
+                error_message = "Invalid adapter strategy"
+                raise ValueError(error_message)
+
+        # Activate the sub-adapter with the selected rank
+        for lora_quantizer in lora_quantizers:
+            lora_quantizer.set_active_rank(selected_rank)
+        activated_rank_config.append(selected_rank)
+
+    return activated_rank_config
+
+
+@torch.no_grad()
+def export_to_openvino(
+    pretrained: str,
+    ckpt_file: Path,
+    ir_dir: Path,
+    specific_rank_config: list[int] = None,
+) -> OVModelForCausalLM:
+    """
+    Create a wrapper of OpenVINO model from the checkpoint for evaluation on CPU.
+
+    :param pretrained: The name or path of the pretrained model.
+    :param ckpt_file: The path to the checkpoint file to load the model weights and NNCF configurations.
+    :param ir_dir: The directory where the OpenVINO model will be saved.
+    :param specific_rank_config: A specific configuration of ranks for each layer (only needed if NLS is enabled).
+    :return: A wrapper of OpenVINO model ready for evaluation.
+    """
+    model_to_eval = AutoModelForCausalLM.from_pretrained(pretrained, torch_dtype=torch.float32, device_map="cpu")
+    example_input = model_to_eval.dummy_inputs
+    model_to_eval = load_checkpoint(model_to_eval, ckpt_file)
+    if specific_rank_config is not None:
+        configure_lora_adapters(
+            get_layer_id_vs_lora_quantizers_map(model_to_eval),
+            specific_rank_config=specific_rank_config,
+        )
+    model_to_eval = nncf.strip(model_to_eval, do_copy=False, strip_format=StripFormat.DQ, example_input=example_input)
+    export_from_model(model_to_eval, ir_dir, device="cpu")
+    return OVModelForCausalLM.from_pretrained(
+        model_id=ir_dir,
+        trust_remote_code=True,
+        load_in_8bit=False,
+        compile=True,
+        ov_config={"KV_CACHE_PRECISION": "f16", "DYNAMIC_QUANTIZATION_GROUP_SIZE": "0"},
+    )
+
+
+def get_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(add_help=True)
+
+    # Model params
+    parser.add_argument(
+        "--pretrained",
+        type=str,
+        default="Qwen/Qwen2.5-3B-Instruct",
+        help="The model id or path of a pretrained HF model configuration.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=Path,
+        default="output",
+        help="Path to the directory for storing logs, tuning checkpoint, compressed model, validation references.",
+    )
+    parser.add_argument(
+        "--resume",
+        action="store_true",
+        help="Whether to start from previously saved checkpoint. If not specified or checkpoint does not exist, "
+        "start from scratch by post-training weight compression initialization.",
+    )
+    parser.add_argument(
+        "--do_train",
+        action="store_true",
```

Comment:
By default, there's no training but evaluation of custom config. 
Since it's a tuning sample, please make do_train=True, by default

Reply:
@ljaljushkin Fixed. do_train is by default true. Thanks!